classes/class-woothemes-sensei-certificate-data-cleaner.php

```diff
@@ -0,0 +1,294 @@
+<?php
+/**
+ * Defines a class with methods for cleaning up plugin data. To be used when
+ * the plugin is deleted.
+ *
+ * @package Core
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	// Exit if accessed directly.
+	exit;
+}
+
+/**
+ * Class for cleaning up all plugin data.
+ *
+ * @author Automattic
+ * @since  2.2.2
+ */
+class WooThemes_Sensei_Certificate_Data_Cleaner {
+
+	/**
+	 * Custom post types to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $custom_post_types = array(
+		'certificate',
+		'certificate_template',
+	);
+
+	/**
+	 * Options to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $options = array(
+		'sensei_certificates_version',
+		'sensei_certificate_user_data_installer',
+		'sensei_certificate_templates_installer',
+		'sensei_certificates_view_certificate_button_added',
+	);
+
+	/**
+	 * Attachment GUIDs (as MySQL regexes) to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $attachment_guids = array(
+		'.+sensei_certificate_nograde.png$',
```

Comment:
Is this the default certificate template image, or something else?

Reply:
Yes, this is the default certificate template image. It's created as an attachment once the plugin is installed. I've decided to clean it up because if the user uses the uninstall feature and reinstalls, the image is created again. If we don't want to remove the image, a possible solution would be to check if an image with that guide exists before creating it.

Reply:
I'm good with cleaning up after ourselves as much as possible. Was mostly just curious. 🙂